library/core/src/iter/adapters/take.rs

```diff
@@ -100,6 +99,26 @@ where
 
     impl_fold_via_try_fold! { fold -> try_fold }
 
+    #[inline]
+    fn for_each<F: FnMut(Self::Item)>(mut self, f: F) {
+        // The default implementation would use a unit accumulator, so we can
+        // avoid a stateful closure by folding over the remaining number
+        // of items we wish to return instead.
+        fn check<'a, Item>(
+            mut action: impl FnMut(Item) + 'a,
+        ) -> impl FnMut(usize, Item) -> Option<usize> + 'a {
+            move |more, x| {
+                action(x);
+                more.checked_sub(1)
+            }
+        }
```

Comment:
A pity that the duplication-by-unused-generics issues still haven't been fixed 😮‍💨